### PR TITLE
Fix the return value of memcpy_to/from_uspace().

### PR DESCRIPTION
**Pull request by le-jzr**
Other 64b platforms need to have the assembly part reviewed.

### CODE REVIEW

File: kernel/arch/amd64/src/asm.S [hunk 1]
```diff
@@ -78,7 +78,7 @@ FUNCTION_END(memcpy_to_uspace)
 
 SYMBOL(memcpy_from_uspace_failover_address)
 SYMBOL(memcpy_to_uspace_failover_address)
-	xorl %eax, %eax         /* return 0, failure */
```
Comment (jermar) [inline kernel/arch/amd64/src/asm.S]: Shouldn't really be necessary on amd64 as it will zero-extend the result to the full 64-bit register

Comment (le-jzr) [reply kernel/arch/amd64/src/asm.S]: Can't hurt either, though.

File: kernel/generic/src/syscall/copy.c [hunk 2]
```diff
@@ -140,12 +140,12 @@ errno_t copy_to_uspace(void *uspace_dst, const void *src, size_t size)
 	ipl = interrupts_disable();
 	THREAD->in_copy_to_uspace = true;
 
-	rc = memcpy_to_uspace(uspace_dst, src, size);
+	rc = !memcpy_to_uspace(uspace_dst, src, size) ? EPERM : 0;
```
Comment (jermar) [inline kernel/generic/src/syscall/copy.c]: Can this be EOK instead of 0?

**Comment by jermar** [comment]
The other 64-bit platforms look good too. So in fact, no need to change assembly for this PR.

**Pull request closed without merging**